cmd/argo/commands/watch.go

```diff
@@ -41,74 +31,74 @@ func NewWatchCommand() *cobra.Command {
 			ctx, apiClient := client.NewAPIClient(cmd.Context())
 			serviceClient := apiClient.NewWorkflowServiceClient()
 			namespace := client.Namespace()
-			watchWorkflow(ctx, serviceClient, namespace, args[0], getArgs)
+			common.WatchWorkflow(ctx, serviceClient, namespace, args[0], getArgs)
 		},
 	}
-	command.Flags().StringVar(&getArgs.status, "status", "", "Filter by status (Pending, Running, Succeeded, Skipped, Failed, Error)")
-	command.Flags().StringVar(&getArgs.nodeFieldSelectorString, "node-field-selector", "", "selector of node to display, eg: --node-field-selector phase=abc")
+	command.Flags().StringVar(&getArgs.Status, "status", "", "Filter by status (Pending, Running, Succeeded, Skipped, Failed, Error)")
+	command.Flags().StringVar(&getArgs.NodeFieldSelectorString, "node-field-selector", "", "selector of node to display, eg: --node-field-selector phase=abc")
 	return command
 }
 
-func watchWorkflow(ctx context.Context, serviceClient workflowpkg.WorkflowServiceClient, namespace string, workflow string, getArgs getFlags) {
-	req := &workflowpkg.WatchWorkflowsRequest{
-		Namespace: namespace,
-		ListOptions: &metav1.ListOptions{
-			FieldSelector:   util.GenerateFieldSelectorFromWorkflowName(workflow),
-			ResourceVersion: "0",
-		},
-	}
-	stream, err := serviceClient.WatchWorkflows(ctx, req)
-	errors.CheckError(err)
+// func watchWorkflow(ctx context.Context, serviceClient workflowpkg.WorkflowServiceClient, namespace string, workflow string, getArgs common.GetFlags) {
+// 	req := &workflowpkg.WatchWorkflowsRequest{
+// 		Namespace: namespace,
+// 		ListOptions: &metav1.ListOptions{
+// 			FieldSelector:   util.GenerateFieldSelectorFromWorkflowName(workflow),
+// 			ResourceVersion: "0",
+// 		},
+// 	}
+// 	stream, err := serviceClient.WatchWorkflows(ctx, req)
+// 	errors.CheckError(err)
 
-	wfChan := make(chan *wfv1.Workflow)
-	go func() {
-		for {
-			event, err := stream.Recv()
-			if err == io.EOF {
-				log.Debug("Re-establishing workflow watch")
-				stream, err = serviceClient.WatchWorkflows(ctx, req)
-				errors.CheckError(err)
-				continue
-			}
-			errors.CheckError(err)
-			if event == nil {
-				continue
-			}
-			wfChan <- event.Object
-		}
-	}()
+// 	wfChan := make(chan *wfv1.Workflow)
```

Comment:
???

Reply:
Should have been removed in last commit, not showing in local. Will remove.

Reply:
This is showing from an old commit, it is removed on the most recent one.